src/game/objects/player.ts

```diff
@@ -102,5 +117,75 @@ export abstract class Character {
         if (this.eventBlock > 0) {
             this.eventBlock -= 1
         }
+        // 부스터 지속 시간 처리
+        if (this.itemDuration.boost > 0) {
+            this.itemDuration.boost -= 1
+            if (this.itemDuration.boost <= 0) {
+                this.speed -= 4 // 부스터 효과 종료
+                this.itemDuration.boost = 0
+            }
+        }
+
+        // 쉴드 지속 시간 처리
+        if (this.itemDuration.shield > 0) {
+            this.itemDuration.shield -= 1
+            if (this.itemDuration.shield <= 0) {
+                this.protect = 0 // 쉴드 효과 종료
+                this.itemDuration.shield = 0
+            }
+        }
+
+        // 번개 처리
+        if (this.thunderEffect.length > 0) {
+            for (let i = 0; i < this.thunderEffect.length; i++) {
+                this.thunderEffect[i] -= 1
+                if (this.thunderEffect[i] <= 0) {
+                    if (this.protect <= 0) {
+                        this.eventBlock = 2 / updateInterval
+                    }
+                    this.thunderEffect.splice(i, 1)
+                    i--
+                }
+            }
+        }
+    }
```

Comment:
for문 순회하면서 splice 하면 예상치 못한 결과가 나올 수 있습니다!
(순회가 띄엄띄엄 이루어짐)

![image](https://github.com/user-attachments/assets/5db8c86b-b347-4c7c-a946-44e9077ef524)


제거 로직을 추후에 넣어주세요!
```suggestion 
// from GPT

if (this.thunderEffect.length > 0) {
    // 삭제할 인덱스를 저장할 배열
    let indicesToRemove = [];

    // 배열 순회
    for (let i = 0; i < this.thunderEffect.length; i++) {
        this.thunderEffect[i] -= 1;

        if (this.thunderEffect[i] <= 0) {
            if (this.protect <= 0) {
                this.eventBlock = 2 / updateInterval;
            }
            // 삭제할 인덱스를 기록
            indicesToRemove.push(i);
        }
    }

    // 인덱스를 역순으로 제거 (역순으로 해야 인덱스 변경에 영향을 받지 않음)
    for (let i = indicesToRemove.length - 1; i >= 0; i--) {
        let index = indicesToRemove[i];
        this.thunderEffect.splice(index, 1);
    }
}

```

Reply:
너무 대충 생각했다 머쓱;; 수정할겡